src/store/DrawToolsStore.js

```diff
@@ -83,21 +85,53 @@ export default class DrawToolsStore {
                 defaultValue: parameters[key],
                 type: typeMap[key],
             }));
+
+
+        const drawingItem = this.findComputedDrawing(drawing);
+        if (drawingItem) {
+            title = `${drawingItem.prefix ? `${drawingItem.prefix} - ` : ''} ${drawingItem.text}`;
+        }
+
         this.activeDrawing = drawing;
         this.activeDrawing.highlighted = false;
-        this.settingsDialog.title = formatCamelCase(drawing.name);
+        this.settingsDialog.formTitle = t.translate('Result');
+        this.settingsDialog.title = title;
         this.settingsDialog.setOpen(true);
     }
 
     noTool = () => {
         const count = this.stx.drawingObjects.length;
         if ((this.menu.open && this.context) || (!this.isContinuous && this._pervDrawingObjectCount !== count)) {
             this.stx.changeVectorType('');
-            this.computeActiveDrawTools();
+            this.drawingFinished();
         }
         this._pervDrawingObjectCount = count;
     };
 
+    findComputedDrawing = (drawing) => {
+        const group = this.activeToolsGroup.find(drawGroup => drawGroup.key === drawing.name);
+        if (group) {
+            const drawingItem = group.items.find(item => (
+                item.v0 === drawing.v0
+                    && item.v1 === drawing.v1
+                    && item.d0 === drawing.d0
+                    && item.d1 === drawing.d1
+            ));
```

Comment:
@morteza-binary its so so rarely that two drawings has the same coordinate, and also we cannot update the chart active drawing store to manage the grouping and new naming, as a result this come to my mind.